src/PageHeader/PageHeader.tsx

```diff
@@ -435,10 +425,33 @@ const Description: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({childr
   )
 }
 
+export type NavigationProps = {
+  as?: 'nav' | 'div'
+  'aria-label'?: React.AriaAttributes['aria-label']
+  'aria-labelledby'?: React.AriaAttributes['aria-labelledby']
+} & PageHeaderProps
+
 // PageHeader.Navigation: The local navigation area of the header. Visible on all viewports
-const Navigation: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, hidden = false}) => {
+const Navigation: React.FC<React.PropsWithChildren<NavigationProps>> = ({
+  children,
+  sx = {},
+  hidden = false,
+  as,
+  'aria-label': ariaLabel,
+  'aria-labelledby': ariaLabelledBy,
+}) => {
+  if (as === 'nav' && !ariaLabel && !ariaLabelledBy) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'Use `aria-label` or `aria-labelledby` prop to provide an accessible label to the `nav` landmark for assistive technology',
+    )
```

Comment:
It would be great to wrap this in a `__DEV__` block so that this gets stripped in prod builds.

I also added a PR for potential helper to consolidate on these types of warn patterns: https://github.com/primer/react/pull/2901

In this case, it could look like:

```js
warning(
  as === 'nav' && (ariaLabel || ariaLabelledBy),
  'Use `aria-label` or `aria-labelledby` prop to provide an accessible label to the `nav` landmark for assistive technology'
);
```

Curious to hear what you think 🤔

Reply:
Great catch Josh! Thank you 🙌🏻 I wrapped the warning with `__DEV__` for now but also added a todo to use the utility function when it is merged ✨